dora/underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSUnderFileSystemFactory.java

```diff
@@ -44,27 +44,41 @@ public GCSUnderFileSystemFactory() {}
 
   @Override
   public UnderFileSystem create(String path, UnderFileSystemConfiguration conf) {
-    Preconditions.checkNotNull(path, "path");
+    Preconditions.checkNotNull(path, "Unable to create UnderFileSystem instance:"
+        + " URI path should not be null");
 
-    if (conf.getInt(PropertyKey.UNDERFS_GCS_VERSION) == GCS_VERSION_TWO) {
-      try {
-        return GCSV2UnderFileSystem.createInstance(new AlluxioURI(path), conf);
-      } catch (IOException e) {
-        LOG.error("Failed to create GCSV2UnderFileSystem.", e);
-        throw Throwables.propagate(e);
-      }
-    } else {
-      try {
-        return GCSUnderFileSystem.createInstance(new AlluxioURI(path), conf);
-      } catch (ServiceException e) {
-        LOG.error("Failed to create GCSUnderFileSystem.", e);
-        throw Throwables.propagate(e);
+    if (checkGCSCredentials(conf)) {
```

Comment:
this is not correct, we can't check whether this is set or not since only v1 would set access key and if we use v2 credential we would not be able to create ufs

Reply:
OK, I will fix this later.